nix/cli-prisma.nix

```diff
@@ -0,0 +1,48 @@
+{ config, pkgs, self', ... }:
+
+# This is an impure build for prisma/prisma. We need this because of the way we
+# pakage `prisma-fmt-wasm` and the fact that there's no `pnpm2nix`.
+# See https://zimbatm.com/notes/nix-packaging-the-heretic-way for more details
+# on impure builds.
+let
+  fmt-wasm = self'.packages.prisma-fmt-wasm;
+  version = "4.11.0";
+in
+{
+  packages.cli-prisma = pkgs.runCommand "prisma-cli-${version}"
+    {
+      # Disable the Nix build sandbox for this specific build.
+      # This means the build can freely talk to the Internet.
+      __noChroot = true;
+
+      nativeBuildInputs = [
+        pkgs.autoPatchelfHook
+      ];
+
+      buildInputs = [
+        pkgs.nodejs
+      ];
+    }
+    ''
+      # NIX sets this to something that doesn't exist for purity reasons.
+      export HOME=$(mktemp -d)
+
+      # Install prisma locally, and impurely.
+      npm install prisma@${version}
```

Comment:
If we don't build the CLI from source, pnpm is not a problem, and with the only cost of having the package hash in the module, we could make this derivation pure. [The nixpkgs prisma package](https://github.com/NixOS/nixpkgs/blob/4fabf060fa08c6a9276b9d9b96e6bf7e3cff8aab/pkgs/development/node-packages/overrides.nix#L408) works that way.

Reply:
Tried this and paired with @tomhoule for a bit; turns out this is not trivial at all: the archive does NOT include npm dependencies (e.g. `@prisma/prisma-engines`). I assume the `nixpkgs` infrastructure handles this automatically.

So for now, we've managed to localise the impure/not-sandboxed bit to only this package, thus alleviating any potential problems.

As tooling develops and maybe we get better support for `pnpm` in `nix`, we should revisit this bit.